keel-core/src/main/kotlin/com/netflix/spinnaker/keel/api/Locatable.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ *
+ * Copyright 2019 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.netflix.spinnaker.keel.api
+
+/**
+ * An object which is located in an account and region
+ */
+interface Locatable {
+  val locations: Locations<out RegionSpec>
+}
+
+open class Locations<T : RegionSpec>(
+  val accountName: String,
+  val regions: Set<T>
+) {
+  override fun equals(other: Any?): Boolean {
```

Comment:
This can't be a data class because things override it, and equals comparison is wonky if we don't implement an equals function.

Reply:
Nothing extends this. Things extend `RegionSpec` but there's no different implementations of `Locations`. I think it _can_ be a data class.

Reply:
Changing! Great point. This is much nicer.